### PR TITLE
Feat(BE-8): Added route and function to delete a cover letter and to retrier a user's details

### PR DESCRIPTION
**Pull request by Itzadetunji**
# General Changes

-   Added route to delete a cover letter
-   Added function to delete a cover letter

## Developer Notes

The front-end can call this function to delete a cover letter by making a request to the api endpoint

---

## PR Link
https://linear.app/team-chisel-hng9/issue/BE-8/endpoints-to-delete-cover-letters-get-cvs

## Checklist

-   [x] The base branch is set to `dev`
-   [x] The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
-   [x] The Linear issue has been linked to the PR
-   [x] The General Changes section has been filled out
-   [x] Developer Notes have been added (optional)
-   [ ] End-to-end tests(if any) are passing without any errors
-   [x] Code style generally follows existing patterns
-   [ ] New third-party packages, if any, do not introduce potential security threats
-   [x] There are no CI changes, or they have been OK’d by the devops team
-   [x] Code changes have been quality checked in the ephemeral URL
-   [x] Errors are handled using the right error handles
-   [x] The right thing is returned

### CODE REVIEW

File: server/routes/templateRoutes.js [hunk 1]
```diff
@@ -2,17 +2,17 @@ const express = require("express");
...
 	editACoverLetter,
+	getAllCoverLettersByAUser,
+	deleteCoverLetter,
 } = require("../controllers/templateController");
 
 const auth = require("../middleware/authentication");
 
 // Add your template routes below this
 
-router.get("/template/:id", auth, getACoverLetter);
```
Comment (Delight-Fela-Steve) [inline server/routes/templateRoutes.js:14]: Please state why this was deleted

File: server/controllers/templateController.js [hunk 2]
```diff
@@ -91,8 +92,22 @@ const editACoverLetter = async (req, res) => {
...
 };
 
+const deleteCoverLetter = async (req, res) => {
+	const { templateId } = req.params;
+
+	if (!mongoose.Types.ObjectId.isValid(templateId))
+		return res.status(404).json({ message: "This user id is not valid!" });
+
+	const template = await Template.findByIdAndDelete({ id: templateId });
+
```
Comment (Delight-Fela-Steve) [inline server/controllers/templateController.js:102]: Kindly check if the template exists before returning a successful delete.

**Review by Delight-Fela-Steve — CHANGES_REQUESTED** [review]

File: server/routes/templateRoutes.js (hunk 1 above)
Comment (Itzadetunji) [reply server/routes/templateRoutes.js:14]: That has been resolved

File: server/controllers/templateController.js (hunk 2 above)
Comment (Itzadetunji) [reply server/controllers/templateController.js:102]: That has been resolved

**Pull request closed without merging**